### PR TITLE
EAR 957 Set Form type for each theme

### PR DESCRIPTION
**Pull request by jasehumphr**
> BEFORE MAKING YOUR PR... There must be no linting errors, all tests must pass and PR is named after JIRA ticket number e.g. EAR-###.

### What is the context of this PR?

> Describe what you have changed as part of this task and why; include a link to the Trello card/GitHub issue, related pull requests, etc as required.
>
> E.g. _"Currently, if you add a new question to Author the entire app crashed. This is because..."_

### How to review

> Add to the list below as appropriate, including screenshots when necessary

- Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
  - ensure it can be opened in Author;
  - then, ensure it can be viewed in Runner by pressing the **view survey** button
- Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process

### CODE REVIEW

File: eq-author/src/App/settings/ThemesPage.js [hunk 1]
```diff
@@ -217,14 +261,28 @@ const ThemesPage = ({ questionnaire }) => {
...
-                          </Field>
-                          <EqIdInput
-                            eqId={eqId}
-                            questionnaireId={id}
-                            shortName={shortName}
-                          />
+                          <StyledIdContainerOuter>
+                            <StyledEqIdContainer>
+                              <Field>
+                                <Label>eQ ID</Label>
```
Comment (MattElcock) [inline eq-author/src/App/settings/ThemesPage.js:267]:
```suggestion
                                <Label>EQ ID</Label>
```

File: eq-author/src/App/settings/ThemesPage.test.js [hunk 2]
```diff
@@ -363,4 +363,50 @@ describe("Themes page", () => {
...
+
+    const formTypeInput = screen.getByTestId("default-form-type-input");
+
+    fireEvent.change(formTypeInput, {
+      target: { value: " " },
+    });
+
+    fireEvent.blur(formTypeInput);
+    expect(handleFormTypeBlur).not.toHaveBeenCalled();
+  });
```
Comment (MattElcock) [inline eq-author/src/App/settings/ThemesPage.test.js:411]: You could fix the failing test by modifying it to instead assert that white space is stripped out.

```suggestion
  it("Should remove spaces when there are any", () => {
    const handleFormTypeBlur = jest.fn();
    useMutation.mockImplementation(() => [handleFormTypeBlur]);
    renderThemesPage(mockQuestionnaire, user, mocks);

    const formTypeInput = screen.getByTestId("default-form-type-input");

    fireEvent.change(formTypeInput, {
      target: { value: " " },
    });

    fireEvent.blur(formTypeInput);
    
    expect(handleFormTypeBlur).toHaveBeenCalledWith(
      expect.objectContaining({
        variables: {
          input: {
            questionnaireId: expect.any(String),
            formType: "",
            shortName: "default",
          },
        },
      })
    );
  });
```

**Review by MattElcock — CHANGES_REQUESTED** [review]
Hey Jason, nice one on this mate! Code looks nice and tidy and does the job. 

Two small suggestions: one fixing some text as it doesn't match the design exactly, and another is a suggestion on how you could fix the failing test.

File: eq-author/src/App/settings/ThemesPage.js (hunk 1 above)
Comment (jasehumphr) [reply eq-author/src/App/settings/ThemesPage.js:267]: That was Sam 🤣

Comment (farres1) [reply eq-author/src/App/settings/ThemesPage.js]: This was changed after a discussion with Curtis. The intended text is "eQ ID", without the capital.

File: eq-author/src/App/settings/ThemesPage.js [hunk 3]
```diff
@@ -217,14 +261,28 @@ const ThemesPage = ({ questionnaire }) => {
...
-                          </Field>
-                          <EqIdInput
-                            eqId={eqId}
-                            questionnaireId={id}
-                            shortName={shortName}
-                          />
+                          <StyledIdContainerOuter>
+                            <StyledEqIdContainer>
+                              <Field>
+                                <Label>EQ ID</Label>
```
Comment (farres1) [inline eq-author/src/App/settings/ThemesPage.js:267]:
```suggestion
                                <Label>eQ ID</Label>
```

Comment (MattElcock) [inline eq-author/src/App/settings/ThemesPage.js:267]:
```suggestion
                                <Label>eQ ID</Label>
```

**Review by MattElcock — CHANGES_REQUESTED** [review]

**Review by farres1 — APPROVED** [review]

**Review by MattElcock — APPROVED** [review]

**Pull request merged**